python/pyspark/mllib/linalg/__init__.py

```diff
@@ -769,10 +769,10 @@ def __getitem__(self, index):
         if index >= self.size or index < 0:
             raise ValueError("Index %d out of bounds." % index)
 
-        insert_index = np.searchsorted(inds, index)
-        if insert_index >= inds.size:
+        if (inds.size == 0) or (index > inds.item(-1)):
```

Comment:
By itself it is slower than using `__getitem__` but it returns standard Python scalar and overall it should be marginally faster when index is a standard scalar as well. 

If you think  that `inds[-1]` is better it shouldn't really matter. With crude tests I get ~40 ns difference on average (245 ns for `[]` and, 205 ns for `inds`)